srk/src/latticePolyhedron.ml

```diff
@@ -0,0 +1,170 @@
+module V = Linear.QQVector
+module P = Polyhedron
+module L = IntLattice
+module T = Linear.MakeLinearMap(QQ)(Int)(V)(V)
+
+include Log.Make(struct let name = "srk.latticePolyhedron" end)
+
+let map_polyhedron map p =
```

Comment:
Geometrically, what does this do?

Reply:
I don't follow -- does "applies" mean "compute the image of", "compute the inverse image of", something else?  (If a polyhedron lives in Q^n, then its cone of valid inequalities live in Q^{n}* x Q, so [map] must be a linear map Q^{n}* x Q -> Q^{m}* x Q, and I don't see how this should be interpreted.  A possible explanation is what's happening is that we're computing the image of a cone under a map, and that cone happens to be the cone of valid inequalities of a polyhedron, but we shouldn't attempt make sense of this operation on the polyhedra themselves)

Reply:
[map] is an invertible linear map from Q^{n} * Q -> Q^{n} * Q that preserves the constant dimension, and is applied to the cone of valid inequalities of the polyhedron P. When we compute the image of this cone under [map] and interpret it as constraints of a polyhedron Q, i.e., the result of the function, Q = (adjoint(map))^{-1}(P).

This interpretation depends on [map] preserving 1 and being an endomorphism whose image contains P (I think); it certainly works when [map] is invertible and preserves 1.